File: frontend/src/types/analysis.ts
import type { Scenario } from './api'
import type { ReuseSummary } from './recovery'

export type AnalysisEventType = 'NORMAL' | 'PROBABLE_LEAK' | 'SUSPICIOUS_USAGE' | 'WATER_QUALITY_ANOMALY' | 'SENSOR_ANOMALY'
export type AnalysisStatus = 'NORMAL' | 'WARNING' | 'ALERT' | 'QUALITY'

export interface ZoneAnalysis {
  zone: string
  status: AnalysisStatus
  anomaly_score: number
  evidence: string[]
}

export interface CurrentEvent {
  event_type: AnalysisEventType
  affected_zone: string | null
  severity: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
  anomaly_score: number
  severity_reason: string
  explanation: string[]
  recommended_action: string
}

export interface AnalysisResponse {
  current_event: CurrentEvent
  zone_analysis: ZoneAnalysis[]
  reuse?: ReuseSummary | null
}

export interface DetectedEvent extends Omit<CurrentEvent, 'severity_reason'> {
  timestamp: string
}

export type AnalysisScenario = Scenario

File: backend/app/schemas/analysis.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.recovery import ReuseSummary


class ZoneAnalysis(BaseModel):
    zone: str
    status: str
    anomaly_score: float
    evidence: list[str] = Field(default_factory=list)


class CurrentEvent(BaseModel):
    event_type: str
    affected_zone: str | None
    severity: str
    anomaly_score: float
    severity_reason: str
    explanation: list[str]
    recommended_action: str


class AnalysisResponse(BaseModel):
    current_event: CurrentEvent
    zone_analysis: list[ZoneAnalysis]
    reuse: ReuseSummary | None = None


class DetectedEvent(BaseModel):
    timestamp: datetime
    event_type: str
    affected_zone: str | None
    severity: str
    anomaly_score: float
    explanation: list[str]
    recommended_action: str
